feat(numeric): add step support to oui numeric component

radireddy opened the pull request:
Signed-off-by: Ravindra Adireddy <ravindra.adireddy@ovhcloud.com>

### Requirements

* Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
* All new code requires tests to ensure against regressions

## Title of the Pull Requests <!-- required -->
Add step support to the oui-numeric component

### Description of the Change

Add step support to the oui-numeric component

### Benefits

Stepper supports customized step value.

### Possible Drawbacks

none

### Applicable Issues

MANAGER-6799

Commit:
feat(numeric): add step support to oui numeric component

Signed-off-by: Ravindra Adireddy <ravindra.adireddy@ovhcloud.com>

marie-j reviewed (commented):
Changes look ok to me but commit message is confusing imo, why not use `step` instead of `limit` ? 
Also should target `develop` no ?

radireddy commented:
> Changes look ok to me but commit message is confusing imo, why not use `step` instead of `limit` ?
> Also should target `develop` no ?

Thanks. I have corrected the commit message. I will check with @antleblanc with the target branch.

radireddy commented:
> Signed-off-by: Ravindra Adireddy [ravindra.adireddy@ovhcloud.com](mailto:ravindra.adireddy@ovhcloud.com)
> 
> ### Requirements
> * Filling out the template is required. Any pull request that does not include enough information to be reviewed in a timely manner may be closed at the maintainers' discretion.
> * All new code requires tests to ensure against regressions
> 
> ## Title of the Pull Requests
> Add step support to the oui-numeric component
> 
> ### Description of the Change
> Add step support to the oui-numeric component
> 
> ### Benefits
> Stepper supports customized step value.
> 
> ### Possible Drawbacks
> none
> 
> ### Applicable Issues
> none

Added test cases related to the feature and updated documentation.
